components/ws-manager/pkg/manager/testdata/status_workspace_startup_failed.golden

```diff
@@ -20,7 +20,7 @@
             },
             "class": "default"
         },
-        "phase": 6,
+        "phase": 5,
```

Comment:
Slightly concerned over this test case change. It shows that phase would be potentially stuck in "stopping", instead of being "stopped". We might have a regression here where some workspaces would not stop correctly?

Reply:
🤔 I need to double check the `status_workspace_startup_failed.json`.
I just ran the golden file update `go test ./... -update -force`.

Reply:
Okay, the root cause is that the status test tests the output of the function [getWorkspaceStatus](https://github.com/gitpod-io/gitpod/blob/77d57bcb0a08d881070291d0ba8cfa0a8b45fa83/components/ws-manager/pkg/manager/status_test.go#L114) matches the golden file or not, and only executes _once_.

Since we remove changing the state at [here](https://github.com/gitpod-io/gitpod/blob/347af9bff0b870590f4da5b8df49a98b7297800b/components/ws-manager/pkg/manager/status.go#L352-L358). Therefore, the function output of the `phase` field would be `stopping`, not `stopped`.

However, for this case, the code goes into [here](https://github.com/gitpod-io/gitpod/blob/77d57bcb0a08d881070291d0ba8cfa0a8b45fa83/components/ws-manager/pkg/manager/monitor.go#L489-L502), removing the finalizer, and reenter into the `getWorkspaceStatus` at [here](https://github.com/gitpod-io/gitpod/blob/77d57bcb0a08d881070291d0ba8cfa0a8b45fa83/components/ws-manager/pkg/manager/status.go#L371). In the end, the phase would be `stopped`, that's the server receives final state.